drivers/bluetooth/hci/hci_silabs_siwx91x.c

```diff
@@ -12,14 +12,57 @@
 LOG_MODULE_REGISTER(bt_hci_driver_siwg917);
 
 #include "rsi_ble.h"
+#include "rsi_ble_common_config.h"
+#define BLE_RF_POWER_INDEX     0x0006
+#define BT_OP_VS_RF_POWER_MODE BT_OP(BT_OGF_VS, BLE_RF_POWER_INDEX)
+#define BT_LE_MODE             2
 
+static int rsi_bt_driver_send_tx_pwr_vs_cmd(const struct device *dev);
 static void siwx91x_bt_resp_rcvd(uint16_t status, rsi_ble_event_rcp_rcvd_info_t *resp_buf);
 
 struct hci_data {
 	bt_hci_recv_t recv;
 	rsi_data_packet_t rsi_data_packet;
 };
 
+/**
+ * @brief Send RF power mode configuration command to controller
+ * @param dev Pointer to the device structure
+ * @return 0 on success, negative errno on failure
+ */
+static int rsi_bt_driver_send_tx_pwr_vs_cmd(const struct device *dev)
+{
+	struct net_buf *buf;
+	uint8_t le_tx_power_index = RSI_BLE_PWR_INX;
+	uint8_t protocol_mode = BT_LE_MODE;
+	int err;
+
+	/* Allocate HCI command buffer with timeout */
+	buf = bt_hci_cmd_alloc(K_FOREVER);
+	if (!buf) {
+		LOG_ERR("Failed to allocate HCI command buffer");
+		return -ENOMEM;
+	}
+	buf = bt_hci_cmd_alloc(K_FOREVER);
+	if (!buf) {
+		LOG_ERR("Failed to allocate HCI command buffer");
+		return -ENOMEM;
+	}
```

Comment:
Why are you allocating the buffer twice? The second allocation also leaks the memory from the first one.